contracts/src/tests/end_turn.cairo

```diff
@@ -54,3 +47,54 @@ fn test_end_turn() {
     // Check that player energy is correclty incremented at the end of each turn.
     assert(player.remaining_energy == 2, 'Wrong player energy value');
 }
+#[test]
+#[available_gas(30000000)]
+fn test_end_turn_with_card_on_side() {
+    let world = spawn_world();
+    let (player1, player2) = get_players();
+    let game_id = create_game(:world, :player1, :player2);
+    let card = Card {
+        token_id: 1,
+        dribble: 1,
+        current_dribble: 1,
+        defense: 2,
+        current_defense: 2,
+        cost: 1,
+        role: Roles::Goalkeeper,
+        is_captain: false
+    };
+    set!(world, (card));
+    set_caller_address(player1);
+    // 1_u256.low, 0_u256.high, Roles::Defender
+    let place_card_calldata = array![game_id, 1, 0, 1];
+    let player = get!(world, (game_id, player1), Player);
+
+    assert(player.remaining_energy == 1, 'energy should be 1');
+    world.execute('place_card_system', place_card_calldata);
+    let player = get!(world, (game_id, player1), Player);
+
+    assert(player.remaining_energy == 0, 'energy should be 0');
+
+    match player.defender {
+        Option::Some(placement) => {
+            match placement {
+                Placement::Side(id) => assert(id == 1, 'Card id should be 1'),
+                Placement::Field(_) => assert(false, 'Wrong Placement')
+            }
+        },
+        Option::None => assert(false, 'Should be some'),
+    }
+    let end_turn_calldata: Array = array![game_id];
+    world.execute('end_turn_system', end_turn_calldata);
```

Comment:
if player 1 plays card A on turn n the card will go on the field at the beginning of turn n + 2 not turn n +1. A card enters the field at the beginning of the next turn of the player who played it. So here it should still be on the side. You should call it once more to make it go on the field